colin/core/loader.py

```diff
@@ -99,35 +101,40 @@ class CheckLoader(object):
     find recursively all checks on a given path
     """
 
-    def __init__(self, path):
+    def __init__(self, checks_paths):
         """
-        :param path: str, path to a file or a dir where check classes are present
+        :param checks_paths: list of str, directories where the checks are present
         """
-        logger.debug("Will load checks from path '{}'.".format(path))
+        logger.debug("Will load checks from paths '%s'.", checks_paths)
+        for p in checks_paths:
+            if os.path.isfile(p):
+                raise RuntimeError("Provided path %s is not a directory." % p)
         self._check_classes = None
         self._mapping = None
-        self.path = path
-        for p in sys.path:
-            if p in self.path:
-                self.top_py_path = p
-                break
-        else:
-            self.top_py_path = path
-            sys.path.insert(0, path)
-            logger.debug("%s is not on pythonpath, added it", path)
+        self.paths = checks_paths
 
     def obtain_check_classes(self):
         """ find children of AbstractCheck class and return them as a list """
         check_classes = set()
-        if os.path.isfile(self.path):
-            return load_check_classes_from_file(self.path, self.top_py_path)
-        for root, _, files in os.walk(self.path):
-            for fi in files:
-                if not fi.endswith(".py"):
-                    continue
-                path = os.path.join(root, fi)
-                check_classes = check_classes.union(set(
-                    load_check_classes_from_file(path, self.top_py_path)))
+        for path in self.paths:
+            for sys_path in sys.path:
+                if sys_path and sys_path in path:
+                    # this is a directory which:
+                    #   1. has to be on sys.path
+                    #   2. is root of the import sequence
+                    top_py_path = sys_path
```

Comment:
I'm not sure whit this.  I'm little bit against to using this ``top_path`` sequence  and adding to ``sys.path``
Instead of that I would prefer to change method ``_load_module`` to use full path to python file and use ``imp.load_source(mod_name, filepath)`` 
from my PoV, loader can be then simplified to (it is just pseudocode):
```
for path in check_pathes
    for root, _, files in os.walk(path):
        for fi in files:
            check_classes = check_classes.union...(load_check_classes_from_file(fi))
```
where ``load_check_classes_from_file`` return set of ``AbstractCheck`` inherited classes

and we can avoid to modify ``sys.path`` and any ``top_path`` operations

Reply:
The reason it's like this is that I was assuming that the files need to be on pythonpath when being imported.

I really like your suggestion and will try to figure it out.

Reply:
Honzo, I added a new commit which implements your suggestion, please take a look.